### PR TITLE
[Hub Generated] Review request for Microsoft.Security to add version preview/2022-08-01-preview

### PR DESCRIPTION
**Pull request by Ameer-az**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Ameer-az/azure-rest-api-specs/release-security-Microsoft.Security-2022-08-01-preview...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json [hunk 1]
```diff
@@ -0,0 +1,1318 @@
...
+            },
+            "cloudRoleArn": {
+              "type": "string",
+              "description": "The cloud role ARN in AWS for this feature"
+            },
+            "configuration": {
+              "type": "object",
+              "description": "configuration for arc auto-provisioning",
+              "properties": {}
+            }
```
Comment (rkmanda) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:804]: Please provide a definition

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json [hunk 2]
```diff
@@ -0,0 +1,1318 @@
...
+              "description": "Is arc auto provisioning enabled"
+            },
+            "cloudRoleArn": {
+              "type": "string",
+              "description": "The cloud role ARN in AWS for this feature"
+            },
+            "configuration": {
+              "type": "object",
+              "description": "configuration for arc auto-provisioning",
+              "properties": {}
```
Comment (rkmanda) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:931]: >properties [](http://example.com/codeflow?start=15&length=10)

Please provide a definition

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json [hunk 3]
```diff
@@ -0,0 +1,1318 @@
...
+          "description": "The ARC autoprovisioning configuration",
+          "properties": {
+            "enabled": {
+              "type": "boolean",
+              "description": "Is arc auto provisioning enabled"
+            },
+            "configuration": {
+              "type": "object",
+              "description": "Configuration for ARC autoprovisioning",
+              "properties": {}
```
Comment (rkmanda) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:1034]: > "properties": {} [](http://example.com/codeflow?start=13&length=17)

Please provide a definition

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json [hunk 4]
```diff
@@ -0,0 +1,1318 @@
...
+          "description": "The ARC autoprovisioning configuration",
+          "properties": {
+            "enabled": {
+              "type": "boolean",
+              "description": "Is arc auto provisioning enabled"
+            },
+            "configuration": {
+              "type": "object",
+              "description": "configuration for arc auto-provisioning",
+              "properties": {}
```
Comment (rkmanda) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:1122]: >properties [](http://example.com/codeflow?start=15&length=10)

Please provide a definition

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json [hunk 5]
```diff
@@ -0,0 +1,1318 @@
...
+                  ],
+                  "x-ms-enum": {
+                    "name": "scanningMode",
+                    "modelAsString": true
+                  }
+                },
+                "exclusionTags": {
+                  "type": "object",
+                  "description": "VM tags that indicates that VM should not be scanned",
+                  "properties": {}
```
Comment (rkmanda) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:1256]: >"properties": {} [](http://example.com/codeflow?start=18&length=16)

If its just a list of tags why not make it an array of strings?

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json [hunk 6]
```diff
@@ -0,0 +1,1318 @@
...
+    "defenderCspmGcpOffering": {
+      "type": "object",
+      "description": "The CSPM P1 for GCP offering",
+      "x-ms-discriminator-value": "DefenderCspmGcp",
+      "allOf": [
+        {
+          "$ref": "#/definitions/cloudOffering"
+        }
+      ],
+      "properties": {}
```
Comment (rkmanda) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:1273]: >"properties": {} [](http://example.com/codeflow?start=6&length=16)

Provide definition

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json [hunk 7]
```diff
@@ -0,0 +1,1318 @@
...
+    "defenderForDevOpsGithubOffering": {
+      "type": "object",
+      "description": "The Defender for DevOps for Github offering",
+      "x-ms-discriminator-value": "DefenderForDevOpsGithub",
+      "allOf": [
+        {
+          "$ref": "#/definitions/cloudOffering"
+        }
+      ],
+      "properties": {}
```
Comment (rkmanda) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:1284]: >"properties": {} [](http://example.com/codeflow?start=6&length=16)

provide definition

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json [hunk 8]
```diff
@@ -0,0 +1,1318 @@
...
+    "defenderForDevOpsAzureDevOpsOffering": {
+      "type": "object",
+      "description": "The Defender for DevOps for Azure DevOps offering",
+      "x-ms-discriminator-value": "DefenderForDevOpsAzureDevOps",
+      "allOf": [
+        {
+          "$ref": "#/definitions/cloudOffering"
+        }
+      ],
+      "properties": {}
```
Comment (rkmanda) [inline specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:1295]: >properties [](http://example.com/codeflow?start=7&length=10)

provide definition

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json (hunk 5 above)
Comment (eliagrady) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:1256]: exclusionTags is a Key-Value pair property bag.
See how [tags](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v1/types.json#L151) are modelled as an object? same here.

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json (hunk 8 above)
Comment (Ameer-az) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:1295]: properties are meant to allow customers to provide additional information for the offering enablement 
in this case no additional properties are required for the offering enablement. that's why properties are empty.

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json (hunk 6 above)
Comment (Ameer-az) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:1273]: properties are meant to allow customers to provide additional information for the offering enablement 
in this case no additional properties are required for the offering enablement. that's why properties are empty.

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json (hunk 4 above)
Comment (amitronen1) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:1122]: We want to maintain alignment between all "XXX auto provisioning" variations to follow the structure of { enabled, configuration }
in this case the model expects empty object.
it was already published like this on previous versions (please see mdeAutoProvisioning)

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json [hunk 9]
```diff
@@ -0,0 +1,1318 @@
...
+          "$ref": "#/definitions/cloudOffering"
+        }
+      ],
+      "properties": {
+        "vmScanners": {
+          "type": "object",
+          "description": "The Microsoft Defender for Server VM scanning configuration",
+          "properties": {
+            "enabled": {
+              "type": "boolean",
```
[the remark this answers is not in the document]
Comment (eliagrady) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:1231]: it is not modeled as `"Enabled": "true"` - but `"Enabled": true`

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json (hunk 4 above)
Comment (Ameer-az) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json]: we deleted the property "configuration"

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json (hunk 2 above)
Comment (Ameer-az) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json]: we deleted the property "configuration"

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json (hunk 1 above)
Comment (Ameer-az) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:799]: we deleted the property "configuration"

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (rkmanda) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:1211]: yes I got that but its a general recommendation to give preference to enums ahead of booleans for the sake of future extensibility. If you are 100% sure there isnt a third state possible its fine to leave it as a boolean

File: specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json (hunk 5 above)
Comment (rkmanda) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:1236]: if its key value pairs of strings can u define it just like tags 
          "type": "object",
          "additionalProperties": {
            "type": "string"
          },

Comment (rkmanda) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json:1236]: also can u remove the free form object - properties pl?

Comment (Ameer-az) [reply specification/security/resource-manager/Microsoft.Security/preview/2022-08-01-preview/securityConnectors.json]: fixed

**Pull request closed without merging**